lib/nodes/Base.ts

```diff
@@ -38,8 +51,8 @@ export class Base<S extends BlueshellState, E> {
 		}
 
 		try {
+			Base.treePublisher.maybePublishTree(state, event, false);
```

Comment:
if you made a non-static member function `Base.publishResult()` that function could be typesafe and also lazy initialize the static `treePublisher` in a typesafe way. I think that way you could make:

`static treePublisher: TreeNonPublisher | TreePublisher<any, any> = new TreeNonPublisher();`

become:

`static treePublisher: TreePublisher<any, any>;`

Reply:
Don't see better typing afforded beyond typing that the register method already has?